src/index.ts

```diff
@@ -259,5 +261,6 @@ function isNumber(input: RLPInput): input is number {
 
 // Check if an input is a BigNumber
 function isBN(input: RLPInput): input is BN {
-    return !!input.hasOwnProperty('toArray');
+  if (!input) return false;
+  return !!input.hasOwnProperty('toArray');

```

Comment:
Any chance to use `bn.js`'s own method for this? This `toArray` stuff is quite flakey.

Reply:
yeah, yeah i wanted to fix all these comments in separate PR. In this one i want to rewrite tests. There are even more issues that i can see with how we use currently BN.js — i needed to fallback to require which is not really the way its supposed to be. It will be addressed in separate PRs.